### PR TITLE
Add function pointers for accelerated XTS to ltc_cipher_descriptor

### PR DESCRIPTION
**Pull request by jforissier**
Similar to what already exists for other modes.

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

### CODE REVIEW

File: src/modes/xts/xts_decrypt.c [hunk 1]
```diff
@@ -87,24 +88,39 @@ static int tweak_uncrypt(const unsigned char *C, unsigned char *P, unsigned char
...
+	   /* use accelerated decryption for whole blocks */
+	   if ((err = desc->accel_xts_decrypt(ct, pt, lim, tweak, &xts->key1,
+					      &xts->key2) != CRYPT_OK)) {
+	      return err;
+	   }
+	   ct += lim * 16;
+	   pt += lim * 16;
+
+	   /* tweak is encrypted on output */
+	   memcpy(T, tweak, sizeof(T));
```
Comment (sjaeckel) [inline src/modes/xts/xts_decrypt.c]: XMEMCPY

File: src/modes/xts/xts_decrypt.c [hunk 2]
```diff
@@ -87,24 +88,39 @@ static int tweak_uncrypt(const unsigned char *C, unsigned char *P, unsigned char
...
       lim = m;
    } else {
       lim = m - 1;
    }
 
-   for (i = 0; i < lim; i++) {
-      err = tweak_uncrypt(ct, pt, T, xts);
-      ct += 16;
-      pt += 16;
+   desc = &cipher_descriptor[xts->cipher];
```
Comment (sjaeckel) [inline src/modes/xts/xts_decrypt.c]: Ah I don't know what I should say... I generally also like it that way, but somehow I don't think we should change the code style only in one file... so please change the pointer back to how it was before :)

**Comment by sjaeckel** [comment]
Thanks for the PR, generally looks fine for me.
Comments are okay I hope.

**Comment by jforissier** [comment]
@sjaeckel thanks for reviewing the patch. New version pushed.

**Pull request closed without merging**